exec_params:
  go_to:
    go_home:
      location_name: ["home"]

  press_blue_button:
    go_to_blue_button:
      location_name: ["blue_button"]
    sensor_zeroing:
    #todo
    close_gripper:
      skill_type: "robotiq_gripper_move"
      position: 0
      force: 20
      velocity: 100
    go_down:
      skill_type: "cart_move"
      frame_id: "tool0"
      twist_move: [0.0,0.0,0.075,0.0,0.0,0.0]
      move_time: 1.25
    press:
      skill_type: "simple_touch"
      target_frame: "tool"
      goal_twist_frame: "tool"      #"base"
      goal_twist: [0.0,0.0,0.02,0.0,0.0,0.0]
      target_wrench: [5]
      wrench_toll: [0.1]
      wrench_deadband: [0.1]
    go_up:
      skill_type: "cart_move"
      frame_id: "tool0"
      twist_move: [0.0,0.0,-0.075,0.0,0.0,0.0]
      move_time: 1.333

  press_red_button:
    go_to_red_button:
      location_name: ["red_button"]
      sensor_zeroing:
            #todo
    close_gripper:
      skill_type: "robotiq_gripper_move"
      position: 0
      force: 20
      velocity: 100
    go_down:
      skill_type: "cart_move"
      frame_id: "tool0"
      twist_move: [0.0,0.0,0.075,0.0,0.0,0.0]
      move_time: 1.25
    press:
      skill_type: "simple_touch"
      target_frame: "tool"
      goal_twist_frame: "tool"      #"base"
      goal_twist: [0.0,0.0,0.02,0.0,0.0,0.0]
      target_wrench: [5]
      wrench_toll: [0.1]
      wrench_deadband: [0.1]
    go_up:
      skill_type: "cart_move"
      frame_id: "tool0"
      twist_move: [0.0,0.0,-0.05,0.0,0.0,0.0]
      move_time: 2

  pick_and_insert_key:
    sensor_zeroing:
          #todo
    pick:
      go_to_key:
        location_name: ["key"]
      open_gripper:
        skill_type: "robotiq_gripper_move"
        position: 50
        force: 20
        velocity: 100
      go_down:
        skill_type: "cart_move"
        frame_id: "world"
        twist_move: [0.0,0.0,-0.045,0.0,0.0,0.0]
        move_time: 1
      touch_down:
        skill_type: "simple_touch"
        target_frame: "tool"
        goal_twist_frame: "world"      #"base"
        goal_twist: [0.0,0.0,-0.01,0.0,0.0,0.0]
        target_wrench: [5]
        wrench_toll: [0.1]
        wrench_deadband: [0.1]
      go_up:
        skill_type: "cart_move"
        frame_id: "world"
        twist_move: [0.0,0.0,0.011,0.0,0.0,0.0]
        move_time: 1
      close_gripper:
        skill_type: "robotiq_gripper_move"
        position: 0
        force: 100
        velocity: 50
      extract_key:
        skill_type: "cart_move"
        frame_id: "tool0"
        twist_move: [0.0,0.0,-0.05,0.0,0.0,0.0]
        move_time: 2
    insert:
      go_to_key_lock:
        location_name: ["key_lock"]
      go_down:
        skill_type: "cart_move"
        frame_id: "world"
        twist_move: [0.0,0.0,-0.03,0.0,0.0,0.0]
        move_time: 2.2
      touch_down:
        skill_type: "simple_touch"
        target_frame: "tool0"
        goal_twist_frame: "world"      #"base"
        goal_twist: [0.0,0.0,-0.001,0.0,0.0,0.0]
        target_wrench: [20]
        wrench_toll: [0.1]go_up:
  
        wrench_deadband: [0.1]
      allign_key_with_hole:
        skill_type: "cart_move"
        frame_id: "tool0"
        twist_move: [0.0, 0.0, 0.0, 0.0, 0.0, -0.314159265]
        move_time: 5
      insert_key:
        skill_type: "simple_touch"
        target_frame: "tool0"
        goal_twist_frame: "world"      #"base"
        goal_twist: [0.00005,0.0,-0.002,0.0,0.0,0.0]
        target_wrench: [15]
        wrench_toll: [0.1]
        wrench_deadband: [0.1]
      partial_open_gripper:
        skill_type: "robotiq_gripper_move"
        position: 80
        force: 20
        velocity: 100
      rotate_key:
        skill_type: "cart_move"
        frame_id: "tool0"
        twist_move: [0.0, 0.0, 0.0, 0.0, 0.0, 0.42]
        move_time: 3
      open_gripper:
        skill_type: "robotiq_gripper_move"
        position: 100
        force: 20
        velocity: 100
      go_up:
        skill_type: "cart_move"
        frame_id: "world"
        twist_move: [0.0,0.0,0.03,0.0,0.0,0.0]
        move_time: 2

  pick_and_insert_ethernet:
    pick:
      go_to_ethernet:
        location_name: ["ethernet"]
      almost_close_gripper:
        skill_type: "robotiq_gripper_move"
        position: 50
        force: 20
        velocity: 100
      go_down:
        skill_type: "cart_move"
        frame_id: "tool0"
        twist_move: [0.0,0.0,0.02,0.0,0.0,0.0]
        move_time: 5.3
      close_gripper:
        skill_type: "robotiq_gripper_move"
        position: 0
        force: 100
        velocity: 20
      extract_ethernet:
        skill_type: "cart_move"
        frame_id: "tool0"
        twist_move: [0.0,0.0,-0.01,0.0,0.0,0.0]
        move_time: 2.0
    insert:
      go_to_ethernet_lock:
        skill_type: "cart_move"
        frame_id: "board"
        twist_move: [0.0, 0.007, 0.0, 0.0, 0.0, 0.0]  #40.25 distanza ethernets (old = 0.0043)
        move_time: 1
      sensor_zeroing:
        #todo
      approach:
        skill_type: "cart_move"
        frame_id: "board"
        twist_move: [0.010,0.0,0.0,0.0,0.0,0.0]
        move_time: 4.4
      rotation_x:
        skill_type: "cart_move"
        frame_id: "eth_hole"
        twist_move: [0.0,0.0,0.0,0.13,0.0,0.0]
        move_time: 2
      rotation_y:
        skill_type: "cart_move"
        frame_id: "eth_hole"
        twist_move: [0.0,0.0,0.0,0.0,0.13,0.0]
        move_time: 2
      go_down:
        skill_type: "cart_move"
        frame_id: "eth_hole"
        twist_move: [0.0,0.0,0.01,0.0,0.0,0.0]
        move_time: 2.5
      touch_z:
        skill_type: "simple_touch"
        target_frame: "ethernet"
        goal_twist_frame: "eth_hole"      #"base"
        goal_twist: [0,0,0.001,0,0,0]
        target_wrench: [6] #era 3
        wrench_toll: [0.1]
        wrench_deadband: [0.1]
      touch_y:
        skill_type: "simple_touch"
        target_frame: "ethernet"
        goal_twist_frame: "eth_hole"      #"base"
        goal_twist: [0.0,0.001,0.0,0.0,0.0,0.0]
        target_wrench: [0, -2, 0, 0, 0, 0] #3
        wrench_toll: [0.0,0.1,0.0,0.0,0.0,0.0]
        wrench_deadband: [0.0,0.1,0.0,0.0,0.0,0.0]
      touch_x:
        skill_type: "simple_touch"
        target_frame: "ethernet"
        goal_twist_frame: "eth_hole"      #"base"
        goal_twist: [-0.001,0.0,0.0,0.0,0.0,0.0]
        target_wrench: [2 , 0, 0, 0, 0, 0] #3
        wrench_toll: [0.1,0.0,0.0,0.0,0.0,0.0]
        wrench_deadband: [0.1,0.0,0.0,0.0,0.0,0.0]
      move_z:
        skill_type: "cart_move"
        frame_id: "ethernet"
        twist_move: [0.0, 0.0, 0.002, 0.0, 0.0 ,0.0]
        move_time: 2
      counter_rotation_x:
        skill_type: "cart_move"
        frame_id: "eth_hole"
        twist_move: [0.0,0.0,0.0,-0.13,0.0,0.0]
        move_time: 2
      counter_rotation_y:
        skill_type: "cart_move"
        frame_id: "eth_hole"
        twist_move: [0.0,0.0,0.0,0.0,-0.13,0.0]
        move_time: 2
      final_touch:
        skill_type: "simple_touch"
        target_frame: "ethernet"
        goal_twist_frame: "eth_hole"      #"base"
        goal_twist: [0,0,0.001,0,0,0]
        target_wrench: [35]
        wrench_toll: [0.1]
        wrench_deadband: [0.1]
      open_gripper:
        skill_type: "robotiq_gripper_move"
        position: 100
        force: 20
        velocity: 100
      go_up:
        skill_type: "cart_move"
        frame_id: "tool0"
        twist_move: [0.0,0.0,-0.02,0.0,0.0,0.0]
        move_time: 3.0

  extract_batteries_AA:
    go_to_batteries_AA_case:
      location_name: ["batteries_AA_case"]
    close_gripper:
      skill_type: "robotiq_gripper_move"
      position: 0
      force: 20
      velocity: 100
    sensor_zeroing:
      #todo
    go_down:
      skill_type: "cart_move"
      frame_id: "tool0"
      twist_move: [0.0,0.0,0.05,0.0,0.0,0.0]
      move_time: 1.6
    touch_case:
      skill_type: "simple_touch"
      target_frame: "tool"
      goal_twist_frame: "tool"      #"base"
      goal_twist: [0.0,0.0,0.003,0.0,0.0,0.0]
      target_wrench: [20]
      wrench_toll: [0.1]
      wrench_deadband: [0.1]
    remove_case:
      skill_type: "cart_move"
      frame_id: "tool0"
      twist_move: [0.0,-0.01,0.0,0.0,0.0,0.0]
      move_time: 7.0
    go_up:
      skill_type: "cart_move"
      frame_id: "tool0"
      twist_move: [0.0,0.0,-0.05,0.0,0.0,0.0]
      move_time: 1.0
